openapi3filter/req_resp_decoder.go

```diff
@@ -526,10 +526,107 @@ func (d *urlValuesDecoder) DecodeArray(param string, sm *openapi3.SerializationM
 		}
 		values = strings.Split(values[0], delim)
 	}
-	val, err := parseArray(values, schema)
+	val, err := d.parseArray(values, sm, schema)
 	return val, ok, err
 }
 
+// parseArray returns an array that contains items from a raw array.
+// Every item is parsed as a primitive value.
+// The function returns an error when an error happened while parse array's items.
+func (d *urlValuesDecoder) parseArray(raw []string, sm *openapi3.SerializationMethod, schemaRef *openapi3.SchemaRef) ([]interface{}, error) {
+	var value []interface{}
+
+	for i, v := range raw {
+		item, err := d.parseValue(v, schemaRef.Value.Items)
+		if err != nil {
+			if v, ok := err.(*ParseError); ok {
+				return nil, &ParseError{path: []interface{}{i}, Cause: v}
+			}
+			return nil, fmt.Errorf("item %d: %w", i, err)
+		}
+
+		// If the items are nil, then the array is nil. There shouldn't be case where some values are actual primitive
+		// values and some are nil values.
+		if item == nil {
+			return nil, nil
+		}
+		value = append(value, item)
+	}
+	return value, nil
+}
+
+func (d *urlValuesDecoder) parseValue(v string, schema *openapi3.SchemaRef) (interface{}, error) {
+	var found bool
+
+	if len(schema.Value.AllOf) > 0 {
+		var value interface{}
+		var err error
+		for _, sr := range schema.Value.AllOf {
+			var f bool
+			value, err = d.parseValue(v, sr)
+			found = found || f
+			if value == nil || err != nil {
+				break
+			}
+		}
+		return value, err
```

Comment:
Returning here implies allOf and siblings are mutually exclusive. They aren't.

Reply:
I follow the same logic as in decodeValue function (https://github.com/getkin/kin-openapi/blob/871e029d62a4d47239d11006431407e08edb1259/openapi3filter/req_resp_decoder.go#L262)

The purpose of this function is to decode the value according to the schema type.
And not to validate it.

If the decoding is successful, then that's it. There is no need to further process the value.
If the value does not match the rest of the siblings, then it will fail in the validation